core/src/main/java/org/mskcc/cbio/portal/authentication/ldap/LDAPUserDetailsContextMapper.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.ldap.DefaultSpringSecurityContextSource;
+import org.springframework.security.ldap.SpringSecurityLdapTemplate;
+import org.springframework.security.ldap.userdetails.UserDetailsContextMapper;
+
+import com.google.inject.internal.Lists;
+
+/**
+ * Mapping between details from an LDAP server and the user name used for authentication.
+ *
+ * Pull first name, given name, and email from the LDAP server.
+ *
+ * @author Manuel Holtgrewe <manuel.holtgrewe@bihealth.de>
+ */
+public class LDAPUserDetailsContextMapper implements UserDetailsContextMapper {
+
+    /** {@link Log} to use */
+    private static final Log LOG = LogFactory.getLog(LDAPUserDetailsContextMapper.class);
+
+    /** {@link DefaultSpringSecurityContextSource} with the LDAP connection */
+    private DefaultSpringSecurityContextSource ldapServer;
+    /** data access object to use for retrieving the authorities from the database */
+    private PortalUserDAO portalUserDAO;
+
+    /** base path to search users in */
+    private String baseDn;
+    /** attribute name to use for finding users in directory */
+    private String usernameAttribute = "sAMAccountName";
+    /** attribute name of the first / given name in directory */
+    private String givenNameAttribute = "givenName";
+    /** attribute name of the last name in the directory */
+    private String lastNameAttribute = "sn";
+    /** attribute name of the email address in the directory */
+    private String emailAttribute = "mail";
+
+    /** Default constructor */
+    public LDAPUserDetailsContextMapper() {
+    }
+
+    /**
+     * Initialize the {@link LDAPUserDetailsContextMapper} with the given {@link DefaultSpringSecurityContextSource} and
+     * {@link PortalUserDAO}.
+     */
+    public LDAPUserDetailsContextMapper(DefaultSpringSecurityContextSource ldapServer, PortalUserDAO portalUserDAO) {
+        this.ldapServer = ldapServer;
+        this.portalUserDAO = portalUserDAO;
+    }
+
+    @Override
+    public UserDetails mapUserFromContext(DirContextOperations ctx, String username,
+            Collection<? extends GrantedAuthority> authorities) {
+        // Query LDAP server for user details, including first/given name and email.
+        SpringSecurityLdapTemplate ldapTemplate = new SpringSecurityLdapTemplate(ldapServer);
+        // ArrayList<String> attributesToRetrieve = Lists.newArrayList("givenname", "sn", "mail");
+        String query = MessageFormat.format("({0}='{'0'}')", usernameAttribute);
+        DirContextOperations user = ldapTemplate.searchForSingleEntry(baseDn, query, new String[] { username });
+        String givenName = (String) user.getAttributeSortedStringSet(givenNameAttribute).first();
+        String lastName = (String) user.getAttributeSortedStringSet(lastNameAttribute).first();
+        String email = (String) user.getAttributeSortedStringSet(emailAttribute).first();
```

Comment:
this will give a NullPointerException when the attribute is not present in the LDAP entry...better exception handling is desired here.